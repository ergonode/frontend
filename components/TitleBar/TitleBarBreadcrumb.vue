/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
<template>
    <NuxtLink
        class="breadcrumb"
        :to="breadcrumb.route">
        <Component
            :is="breadcrumbIconComponent"
            :fill-color="greenColor"
            size="16" />
        <span
            class="breadcrumb__title font--medium-14-20"
            v-text="breadcrumb.title" />
    </NuxtLink>
</template>

<script>
import { GREEN } from '~/assets/scss/_variables/_colors.scss';

export default {
    name: 'TitleBarBreadcrumb',
    props: {
        breadcrumb: {
            type: Object,
            required: true,
        },
    },
    computed: {
        greenColor() {
            return GREEN;
        },
        breadcrumbIconComponent() {
            return () => import(`~/components/Icon/Menu/Icon${this.breadcrumb.icon}`);
        },
    },
};
</script>

<style lang="scss" scoped>
    .breadcrumb {
        display: flex;
        align-items: center;
        grid-column-gap: 6px;
        grid-column: 2 / 3;
        grid-row: 2 / 3;
        margin-left: 8px;
        text-decoration: none;

        &__title {
            margin-left: 4px;
            color: $GRAPHITE_DARK;
        }

        &:hover {
            text-decoration: underline;
        }
    }
</style>
