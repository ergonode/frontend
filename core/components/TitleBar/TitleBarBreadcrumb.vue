/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
<template>
    <NuxtLink
        class="breadcrumb"
        :to="breadcrumb.route">
        <Component
            :is="breadcrumbIconComponent"
            :fill-color="greenColor"
            size="16" />
        <span
            class="breadcrumb__title"
            v-text="breadcrumb.title" />
    </NuxtLink>
</template>

<script>
import { GREEN } from '~/assets/scss/_variables/_colors.scss';

export default {
    name: 'TitleBarBreadcrumb',
    props: {
        breadcrumb: {
            type: Object,
            required: true,
        },
    },
    computed: {
        greenColor() {
            return GREEN;
        },
        breadcrumbIconComponent() {
            return () => import(`~/components/Icon/Menu/Icon${this.breadcrumb.icon}`);
        },
    },
};
</script>

<style lang="scss" scoped>
    .breadcrumb {
        display: flex;
        align-items: center;
        grid-column: 2 / 3;
        grid-row: 2 / 3;
        margin-left: 8px;
        text-decoration: none;

        &:active {
            color: inherit;
        }

        &__title {
            margin-left: 4px;
            color: $GRAPHITE_DARK;
            font: $FONT_MEDIUM_14_20;
        }

        &:hover {
            text-decoration: underline;
        }
    }
</style>
