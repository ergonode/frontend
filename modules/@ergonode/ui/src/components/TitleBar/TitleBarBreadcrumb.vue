/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
<template>
    <NuxtLink
        class="breadcrumb"
        :to="breadcrumb.route">
        <Component
            :is="breadcrumbIconComponent"
            :fill-color="greenColor"
            width="16"
            height="16" />
        <span
            class="breadcrumb__title"
            v-text="breadcrumb.title" />
    </NuxtLink>
</template>

<script>
import {
    GREEN,
} from '@UI/assets/scss/_js-variables/colors.scss';

export default {
    name: 'TitleBarBreadcrumb',
    props: {
        /**
         * Breadcrumb data model
         */
        breadcrumb: {
            type: Object,
            required: true,
        },
    },
    computed: {
        greenColor() {
            return GREEN;
        },
        breadcrumbIconComponent() {
            return () => import(`@Core/components/Icons/Menu/Icon${this.breadcrumb.icon}`);
        },
    },
};
</script>

<style lang="scss" scoped>
    .breadcrumb {
        display: flex;
        align-items: center;
        grid-column: 2 / 3;
        grid-row: 2 / 3;
        margin-left: 8px;
        text-decoration: none;

        &:active {
            color: inherit;
        }

        &__title {
            margin-left: 4px;
            color: $GRAPHITE_DARK;
            font: $FONT_MEDIUM_14_20;
        }

        &:hover {
            text-decoration: underline;
        }
    }
</style>
