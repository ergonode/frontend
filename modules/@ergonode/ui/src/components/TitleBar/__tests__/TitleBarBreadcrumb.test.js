/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
import TitleBarBreadcrumb from '@UI/components/TitleBar/TitleBarBreadcrumb';
import {
    createLocalVue,
    shallowMount,
} from '@vue/test-utils';

const localVue = createLocalVue();

describe('TitleBar/TitleBarBreadcrumb', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallowMount(TitleBarBreadcrumb, {
            localVue,
            propsData: {
                breadcrumb: {
                    title: 'Categories',
                    icon: 'Category',
                    path: '/categories',
                },

            },
            stubs: {
                NuxtLink: true,
            },
        });
    });

    it('Component is named well', () => {
        expect(typeof TitleBarBreadcrumb.name).toBe('string');
        expect(TitleBarBreadcrumb.name).toEqual('TitleBarBreadcrumb');
    });

    it('Renders breadcrumb with category', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
