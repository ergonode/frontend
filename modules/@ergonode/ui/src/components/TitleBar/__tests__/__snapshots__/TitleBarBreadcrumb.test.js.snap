// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleBar/TitleBarBreadcrumb Renders breadcrumb with category 1`] = `
<nuxtlink-stub class="breadcrumb">
  <!----> <span class="breadcrumb__title">Categories</span>
</nuxtlink-stub>
`;
